esphome/components/alarm_control_panel/__init__.py

```diff
@@ -0,0 +1,184 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.automation import maybe_simple_id
+from esphome.components import binary_sensor
+from esphome.core import CORE, coroutine_with_priority
+from esphome.const import (
+    CONF_ID,
+    CONF_BINARY_SENSORS,
+    CONF_INPUT,
+    CONF_ON_STATE,
+    CONF_TRIGGER_ID,
+    CONF_CODE,
+)
+from esphome.cpp_helpers import setup_entity
+
+CODEOWNERS = ["@grahambrown11"]
+DEPENDENCIES = ["binary_sensor"]
+
+CONF_CODES = "codes"
+CONF_BYPASS_ARMED_HOME = "bypass_armed_home"
+CONF_REQUIRES_CODE_TO_ARM = "requires_code_to_arm"
+CONF_ARMING_HOME_TIME = "arming_home_time"
+CONF_ARMING_AWAY_TIME = "arming_away_time"
+CONF_DELAY_TIME = "delay_time"
+CONF_TRIGGER_TIME = "trigger_time"
+CONF_ON_TRIGGERED = "on_triggered"
+CONF_ON_CLEARED = "on_cleared"
+
+AlarmPanelNS = cg.esphome_ns.namespace("alarm_control_panel")
+AlarmControlPanelComponent = AlarmPanelNS.class_("AlarmControlPanel", cg.Component)
```

Comment:
Variable name should match the actual class name please.
```suggestion
AlarmControlPanel = AlarmPanelNS.class_("AlarmControlPanel", cg.Component)
```